Correct BigQuery partitioning/clustering metadata in ETL `metadata.yaml` files

sean-rose opened the pull request:
- 135 ETLs (39%) are missing BigQuery partitioning/clustering metadata.
- 4 ETLs have incorrect BigQuery partitioning/clustering metadata.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

whd reviewed (approved):
Regarding only the `search_terms*` related changes: they are no-op and not subject to additional change control

sean-rose commented:
For posterity, here's the Python script I used to update the `metadata.yaml` files, which is adapted from the [`bigquery_etl.cli.query._update_query_schema` method](https://github.com/mozilla/bigquery-etl/blob/6978139496b34b1a6f2f489c53c16048100bcc83/bigquery_etl/cli/query.py#L1436-L1474) because I only wanted to modify `metadata.yaml` files and not `schema.yaml` files.  I ran the script and then (very tediously) manually reverted all the unrelated modifications it made the `metadata.yaml` files (reordering fields, reformatting multi-line strings, removing comments, etc.) to only keep the BigQuery metadata-related changes.
```python
from itertools import chain
from pathlib import Path

from google.cloud import bigquery
import google.cloud.exceptions

from bigquery_etl.metadata.parse_metadata import METADATA_FILE, Metadata


client = bigquery.Client()

sql_dir = (Path(__file__).parent / 'sql')
query_file_paths = chain(
    sql_dir.rglob('query.*'),
    sql_dir.rglob('part1.sql')
)

for query_file_path in query_file_paths:
    query_dir = query_file_path.parent
    dataset_dir = query_dir.parent
    project_dir = dataset_dir.parent

    table_name = query_dir.name
    dataset_name = dataset_dir.name
    project_name = project_dir.name

    try:
        metadata: Metadata = Metadata.of_query_file(str(query_file_path))
        table = client.get_table(f"{project_name}.{dataset_name}.{table_name}")
        metadata_file_path = query_file_path.parent / METADATA_FILE
        updated = False

        if (
            table.time_partitioning
            and (
                metadata.bigquery is None
                or metadata.bigquery.time_partitioning is None
                or metadata.bigquery.time_partitioning.field != table.time_partitioning.field
                or metadata.bigquery.time_partitioning.type.bigquery_type != table.time_partitioning.type_
                or bool(metadata.bigquery.time_partitioning.require_partition_filter) != bool(table.time_partitioning.require_partition_filter)
                or metadata.bigquery.time_partitioning.expiration_ms != table.time_partitioning.expiration_ms
            )
        ):
            metadata.set_bigquery_partitioning(
                field=table.time_partitioning.field,
                partition_type=table.time_partitioning.type_.lower(),
                required=table.time_partitioning.require_partition_filter,
                expiration_days=table.time_partitioning.expiration_ms / 86400000.0
                if table.time_partitioning.expiration_ms
                else None,
            )
            print(f"Partitioning metadata added to {metadata_file_path}")
            updated = True

        if (
            table.clustering_fields
            and (
                metadata.bigquery is None
                or metadata.bigquery.clustering is None
                or metadata.bigquery.clustering.fields != table.clustering_fields
            )
        ):
            metadata.set_bigquery_clustering(table.clustering_fields)
            print(f"Clustering metadata added to {metadata_file_path}")
            updated = True

        if updated:
            metadata.write(metadata_file_path)
    except FileNotFoundError:
        print(f"No metadata defined for {query_file_path}")
    except google.cloud.exceptions.NotFound:
        print(f"Destination table {project_name}.{dataset_name}.{table_name} does not exist in BigQuery")
```

scholtzan commented on sql/moz-fx-data-shared-prod/telemetry_derived/events_1pct_v1/metadata.yaml:
```diff
@@ -13,3 +13,13 @@ scheduling:
+    type: day
+    field: submission_date
+    require_partition_filter: true
+    expiration_days: 180.0
```
should we change these to integers?

scholtzan reviewed (approved):
Aside from `expiration_days` being floats, this looks good to me
